src/control/cmd/daos/pool.go

```diff
@@ -329,6 +330,58 @@ func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.Poo
 		if queryMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
 			poolInfo.DisabledRanks = rs
 		}
+		if queryMask.HasOption(daos.PoolQueryOptionSuspectEngines) {
+			poolInfo.SuspectRanks = rs
+		}
+	}
+
+	return poolInfo, nil
+}
+
+func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.PoolInfo, error) {
+	origQueryMask := queryMask
+	queryMask.ClearOptions(daos.PoolQueryOptionEnabledEngines,
+		daos.PoolQueryOptionSuspectEngines, daos.PoolQueryOptionDisabledEngines)
+	if origQueryMask.HasOption(daos.PoolQueryOptionEnabledEngines) {
+		queryMask.SetOptions(daos.PoolQueryOptionEnabledEngines)
+		origQueryMask.ClearOptions(daos.PoolQueryOptionEnabledEngines)
+	} else if origQueryMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
+		queryMask.SetOptions(daos.PoolQueryOptionDisabledEngines)
+		origQueryMask.ClearOptions(daos.PoolQueryOptionDisabledEngines)
+	} else if origQueryMask.HasOption(daos.PoolQueryOptionSuspectEngines) {
+		queryMask.SetOptions(daos.PoolQueryOptionSuspectEngines)
+		origQueryMask.ClearOptions(daos.PoolQueryOptionSuspectEngines)
+	}
+	poolInfo, err := queryPoolHelper(poolHdl, queryMask)
+	if err != nil {
+		return nil, err
+	}
+	if origQueryMask.HasOption(daos.PoolQueryOptionEnabledEngines) {
+		queryMask.ClearAll()
+		queryMask.SetOptions(daos.PoolQueryOptionEnabledEngines)
+		poolInfo1, err := queryPoolHelper(poolHdl, queryMask)
+		if err != nil {
+			return nil, err
+		}
+		poolInfo.EnabledRanks = poolInfo1.EnabledRanks
+	}
+	if origQueryMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
+		queryMask.ClearAll()
+		queryMask.SetOptions(daos.PoolQueryOptionDisabledEngines)
+		poolInfo1, err := queryPoolHelper(poolHdl, queryMask)
+		if err != nil {
+			return nil, err
+		}
+		poolInfo.DisabledRanks = poolInfo1.DisabledRanks
+	}
+	if origQueryMask.HasOption(daos.PoolQueryOptionSuspectEngines) {
+		queryMask.ClearAll()
+		queryMask.SetOptions(daos.PoolQueryOptionSuspectEngines)
+		poolInfo1, err := queryPoolHelper(poolHdl, queryMask)
+		if err != nil {
+			return nil, err
+		}
+		poolInfo.SuspectRanks = poolInfo1.SuspectRanks
```

Comment:
This seems like it could be a loop or a helper function, as these three blocks are very similar.

Reply:
Sure.